diff --git a/lib/portage/sync/modules/git/git.py b/lib/portage/sync/modules/git/git.py
index 98670e1..2f72510 100644
--- a/lib/portage/sync/modules/git/git.py
+++ b/lib/portage/sync/modules/git/git.py
@@ -221,8 +221,8 @@ class GitSync(NewBase):
 
         if shallow:
             # Since the default merge strategy typically fails when
-            # the depth is not unlimited, `git reset --merge`.
-            merge_cmd = [self.bin_command, "reset", "--merge"]
+            # the depth is not unlimited, `git reset --hard`.
+            merge_cmd = [self.bin_command, "reset", "--hard"]
         else:
             merge_cmd = [self.bin_command, "merge"]
         merge_cmd.append("refs/remotes/%s" % remote_branch)
